python/lsst/summit/utils/guiders/transformation.py

```diff
@@ -0,0 +1,714 @@
+# This file is part of summit_utils.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = [
+    "mk_rot",
+    "mk_ccd_to_dvcs",
+    "mk_roi_bboxes",
+    "stamp_to_ccd",
+    "get_detector_amp",
+    "convert_roi",
+    "focal_to_pixel",
+    "pixel_to_focal",
+    "amp_to_ccdview",
+    "convert_focal_to_altaz",
+    "convert_pixels_to_altaz",
+    "convert_to_focal_plane",
+    "convert_roi_to_ccd",
+    "convert_ccd_to_roi",
+    "convert_ccd_to_dvcs",
+]
+
+from typing import TYPE_CHECKING, Any
+
+import astropy.units as u
+import numpy as np
+from astropy.coordinates import AltAz, SkyCoord
+from astropy.time import Time
+
+from lsst.afw import cameraGeom
+from lsst.afw.cameraGeom import Detector
+from lsst.afw.image import ImageF
+from lsst.geom import AffineTransform, Box2D, Box2I, Extent2D, Point2D
+from lsst.obs.lsst import LsstCam  # noqa F401
+from lsst.obs.lsst.cameraTransforms import LsstCameraTransforms  # noqa F401
+from lsst.obs.lsst.translators.lsst import SIMONYI_LOCATION  # noqa F401
+
+if TYPE_CHECKING:
+    from astropy.time import Time  # noqa F811
+
+    from lsst.summit.utils.guiders.reading import GuiderData
+
+
+def convert_pixel_to_radec(wcs: Any, x_flat: np.ndarray, y_flat: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Map detector-pixel → ICRS RA/Dec (radians).
+    """
+    return wcs.pixelToSkyArray(x_flat, y_flat)
+
+
+def convert_pixels_to_altaz(
+    wcs: Any, time: Time, x_pix: np.ndarray, y_pix: np.ndarray
```

Comment:
it won't make much difference, but may as well turn all the uses of `Any` in this file into their actual classes e.g. SkyWcs like you have elsewhere